### PR TITLE
Add a `pagefind` boolean flag to user config

### PR DESCRIPTION
**Pull request by delucis**
#### Description

- Closes #952
- Adds a `pagefind` boolean to user config that can be set to `false` to disable Pagefind indexing and skip rendering the default search UI.

### CODE REVIEW

File: .changeset/beige-shoes-whisper.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/starlight': minor
+---
+
+Add configuration option to disable Pagefind
```
Comment (sarah11918) [inline .changeset/beige-shoes-whisper.md:5]:
```suggestion
Adds a configuration option to disable site indexing with Pagefind
```
less harsh wording?

File: docs/src/content/docs/reference/configuration.mdx [hunk 2]
```diff
@@ -411,6 +411,16 @@ When `false`, the colors provided by the active syntax highlighting theme are us
 When using custom themes and setting this to `true`, you must provide at least one dark and one light theme to ensure proper color contrast.
 :::
 
+### `pagefind`
+
+**type:** `boolean`  
+**default:** `true`
+
+Define if Starlight’s default site search provider [Pagefind](https://pagefind.app/) is enabled.
```
Comment (sarah11918) [inline docs/src/content/docs/reference/configuration.mdx:419]:
```suggestion
Define whether Starlight’s default site search provider [Pagefind](https://pagefind.app/) is enabled.
```

File: packages/starlight/utils/user-config.ts [hunk 3]
```diff
@@ -190,6 +190,13 @@ const UserConfigSchema = z.object({
 	 */
 	expressiveCode: ExpressiveCodeSchema(),
 
+	/**
+	 * Define if Starlight’s default site search provider Pagefind is enabled.
```
Comment (sarah11918) [inline packages/starlight/utils/user-config.ts:194]:
```suggestion
	 * Define whether Starlight’s default site search provider Pagefind is enabled.
```

**Review by sarah11918 — COMMENTED** [review]
Docs review! Some comments suggested.

File: packages/starlight/components/Header.astro [hunk 4]
```diff
@@ -8,14 +9,20 @@ import {
 	SocialIcons,
 	ThemeSelect,
 } from 'virtual:starlight/components';
+
+/**
+ * Render the `Search` component if Pagefind is enabled or the default search component has been overridden.
```
Comment (HiDeoo) [inline packages/starlight/components/Header.astro:14]: Should this behavior referenced somewhere like in the search component override reference?

Comment (delucis) [reply packages/starlight/components/Header.astro:14]: Ah, maybe. I initially thought about disabling rendering inside `<Search>` itself, but that would result in some CSS & JS still being loaded even if the UI isn’t, so moved it here. I guess I assumed if you override `Search` you don’t also want your override to be hidden if `pagefind` is disabled.

File: .changeset/beige-shoes-whisper.md (hunk 1 above)
Comment (delucis) [reply .changeset/beige-shoes-whisper.md:5]: Expanded a little more in that case:
```suggestion
Adds a configuration option to disable site indexing with Pagefind and the default search UI
```

File: packages/starlight/components/Header.astro (hunk 4 above)
Comment (delucis) [reply packages/starlight/components/Header.astro:14]: Added a note in accf0a2!

**Review by HiDeoo — APPROVED** [review]
Looks perfect and ready to me 👍

**Review by dreyfus92 — APPROVED** [review]
LGTM ✅

**Comment by kevinzunigacuellar** [comment]
I'm considering an API structure like this:

```js
search: 'pagefind' // or false
```
I find this setup more explicit in its configuration. Additionally, this API structure could potentially pave the way for other official plugins like 'algolia', although that's beyond the scope of this comment.

What are your thoughts on this approach?

**Comment by delucis** [comment]
> this API structure could potentially pave the way for other official plugins

Interesting idea! But I'm pretty confident that we won't have other search providers built-in, so this option would end up having only one possible value, which suggests the Boolean toggle probably works well I think.

Alternative search providers could still be added via the new plugins API but they would then be configured in the plug-in options rather than in the global Starlight config.

**Review by kevinzunigacuellar — APPROVED** [review]
Lively Groove, Tune Mastered!

**Pull request merged**